test/powershell/Modules/Microsoft.PowerShell.Core/Enter-PSHostProcess.Tests.ps1

```diff
@@ -63,72 +87,37 @@ Describe "Enter-PSHostProcess tests" -Tag Feature {
-            # Helper function to get the correct path for the named pipe.
-            function Get-PipePath {
-                param (
-                    $PipeName
-                )
-                if ($IsWindows) {
-                    return "\\.\pipe\$PipeName"
-                }
-                "$([System.IO.Path]::GetTempPath())CoreFxPipe_$PipeName"
-            }
 
+        It "Can enter, exit, and re-enter using CustomPipeName" {
             $pipeName = [System.IO.Path]::GetRandomFileName()
-            $params = @{
-                FilePath = "pwsh"
-                ArgumentList = @("-CustomPipeName",$pipeName)
-                PassThru = $true
-                RedirectStandardOutput = "TestDrive:\pwsh_out.log"
-                RedirectStandardError = "TestDrive:\pwsh_err.log"
-            }
-            $pwsh = Start-Process @params
-
             $pipePath = Get-PipePath -PipeName $pipeName
-        }
+            $pwsh = Start-PSProcess -ArgumentList @("-CustomPipeName",$pipeName)
 
-        AfterAll {
-            $pwsh | Stop-Process
-        }
-
-        It "Can enter using CustomPipeName" -Pending:$true {
-            Wait-UntilTrue { Test-Path $pipePath }
-
-            "Enter-PSHostProcess -CustomPipeName $pipeName -ErrorAction Stop
-            `$pid
-            Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
-        }
-
-        It "Can enter, exit, and re-enter using CustomPipeName" -Pending:$true {
-            Wait-UntilTrue { Test-Path $pipePath }
+            try {
+                Wait-UntilTrue { Test-Path $pipePath }
 
-            "Enter-PSHostProcess -CustomPipeName $pipeName -ErrorAction Stop
-            `$pid
-            Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
+                "Enter-PSHostProcess -CustomPipeName $pipeName -ErrorAction Stop
+                `$pid
+                Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
 
-            "Enter-PSHostProcess -CustomPipeName $pipeName -ErrorAction Stop
-            `$pid
-            Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
+                "Enter-PSHostProcess -CustomPipeName $pipeName -ErrorAction Stop
+                `$pid
+                Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
+            } finally {
+                $pwsh | Stop-Process
+            }
         }
 
         It "Should throw if CustomPipeName does not exist" {
-            Wait-UntilTrue { Test-Path $pipePath }
-
             { Enter-PSHostProcess -CustomPipeName badpipename } | Should -Throw -ExpectedMessage "No named pipe was found with CustomPipeName: badpipename."
         }
 
-        It "Should throw if CustomPipeName is too long on Linux or macOS" {
+        It "Should throw if CustomPipeName is too long on Linux or macOS" -Skip:($IsWindows) {
             $longPipeName = "DoggoipsumwaggywagssmolborkingdoggowithalongsnootforpatsdoingmeafrightenporgoYapperporgolongwatershoobcloudsbigolpupperlengthboy"
```

Comment:
Why is this a thing?
(just wondering)

Reply:
because on unix they're not called "Named Pipes" they're called "UNIX domain sockets" which have a limit on macOS and Linux.

Reply:
I meant `
$longPipeName = "DoggoipsumwaggywagssmolborkingdoggowithalongsnootforpatsdoingmeafrightenporgoYapperporgolongwatershoobcloudsbigolpupperlengthboy"`

Reply:
+1 to generate needed length.

Reply:
I see.